mac may build failed

introduced by: https://github.com/pingcap/tiflash/pull/10364
```   [📡 STDERR] In file included from /Users/pingcap/workspace/bp-tiflash-release-darwin-arm64-9jnqq-build-binaries/source/tiflash/dbms/src/Functions/FunctionsRound.cpp:16:
    [📡 STDERR] /Users/pingcap/workspace/bp-tiflash-release-darwin-arm64-9jnqq-build-binaries/source/tiflash/dbms/src/Functions/FunctionsRound.h:272:24: error: missing 'typename' prior to dependent type name 'T::NativeType'
    [📡 STDERR]     using NativeType = T::NativeType;
    [📡 STDERR]                        ^~~~~~~~~~~~~
    [📡 STDERR]                        typename 
    [📡 STDERR] 1 error generated.```

--- a/dbms/src/Functions/FunctionsRound.h
+++ b/dbms/src/Functions/FunctionsRound.h
@@ -269,7 +269,7 @@ template <typename T, RoundingMode rounding_mode, ScaleMode scale_mode, typename
 struct DecimalRoundingComputation
 {
     static_assert(IsDecimal<T>);
-    using NativeType = T::NativeType;
+    using NativeType = typename T::NativeType;
     static const size_t data_count = 1;
     static size_t prepare(size_t scale) { return scale; }
     // compute need decimal_scale to interpret decimals

